<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Groupbox */

QGroupBox {
    border: 1px solid #DDD;
    border-radius: 9px;
    margin-top: 9px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="SettingsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Settings</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <widget class="QSpinBox" name="SubdivSteps">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>320</y>
         <width>181</width>
         <height>29</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="ImportOBJ">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>181</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Import OBJ file</string>
       </property>
      </widget>
      <widget class="QLabel" name="StepsLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>290</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Catmull-Clark steps</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>90</y>
         <width>87</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Wireframe</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QCheckBox" name="limitPointsCB">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>360</y>
         <width>87</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>To limit</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="quadPatchCB">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>150</y>
         <width>161</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Quad Patches</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="controlMeshCB">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>120</y>
         <width>131</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Control Mesh</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="sharpnessSlider">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>460</y>
         <width>62</width>
         <height>27</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>430</y>
         <width>111</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Sharpness</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="quadPatchGB">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>185</y>
         <width>166</width>
         <height>80</height>
        </rect>
       </property>
       <property name="title">
        <string>Quad Patch Setting</string>
       </property>
       <widget class="QSpinBox" name="outerLevelSB">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>41</width>
          <height>27</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
       <widget class="QSpinBox" name="innerLevelSB">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>41</width>
          <height>27</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_3">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>55</y>
          <width>91</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Innter Level</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_2">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>25</y>
          <width>101</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Outer Level</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="MainDisplay"/>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
